fdbclient/WriteMap.h

```diff
@@ -333,25 +333,27 @@ class WriteMap {
 
 		enum SEGMENT_TYPE { UNMODIFIED_RANGE, CLEARED_RANGE, INDEPENDENT_WRITE, DEPENDENT_WRITE };
 
-		SEGMENT_TYPE type() {
+		SEGMENT_TYPE type() const {
 			if (offset) 
 				return entry().following_keys_cleared ? CLEARED_RANGE : UNMODIFIED_RANGE;
 			else
 				return entry().stack.isDependent() ? DEPENDENT_WRITE : INDEPENDENT_WRITE;
 		}
-		bool is_cleared_range() { return offset && entry().following_keys_cleared; }
-		bool is_unmodified_range() { return offset && !entry().following_keys_cleared; }
-		bool is_operation() { return !offset; }
-		bool is_conflict_range() { return offset ? entry().following_keys_conflict : entry().is_conflict; }
-		bool is_unreadable() { return offset ? entry().following_keys_unreadable : entry().is_unreadable; }
+		bool is_cleared_range() const { return offset && entry().following_keys_cleared; }
+		bool is_unmodified_range() const { return offset && !entry().following_keys_cleared; }
+		bool is_operation() const { return !offset; }
+		bool is_conflict_range() const { return offset ? entry().following_keys_conflict : entry().is_conflict; }
+		bool is_unreadable() const { return offset ? entry().following_keys_unreadable : entry().is_unreadable; }
 
-		bool is_independent() { return entry().following_keys_cleared || !entry().stack.isDependent(); }  // Defined if is_operation()
+		bool is_independent() const {
+			return entry().following_keys_cleared || !entry().stack.isDependent();
+		} // Defined if is_operation()
```

Comment:
Nit: we may want to move this comment, or possibly even consider rewriting it as an assert?